interfaces_state_broadcaster/doc/userdoc.rst

```diff
@@ -0,0 +1,28 @@
+:github_url: https://github.com/ros-controls/ros2_controllers/blob/{REPOS_FILE_BRANCH}/interfaces_state_broadcaster/doc/userdoc.rst
+
+.. _interfaces_state_broadcaster_userdoc:
+
+Interfaces State Broadcaster
+--------------------------------
+The Interfaces State Broadcaster publishes the state of specified hardware interfaces that support double data type.
```

Comment:
This is not 100% true. "is_castable_to_double" is true for all datatypes we currently support, but the get_optional<double> only works for double and bool with additional warning.
We should either limit it to double-only, or support all castable datatypes for real ;)
This is what I started [here](https://github.com/ros-controls/ros2_control/pull/2857/files).